src/PointwiseFunctions/AnalyticSolutions/NewtonianEuler/LaneEmdenStar.cpp

```diff
@@ -0,0 +1,166 @@
+}
+
+void LaneEmdenStar::pup(PUP::er& p) noexcept {
+  p | central_mass_density_;
+  p | polytropic_constant_;
+  p | equation_of_state_;
+  p | source_term_;
+}
+
+template <typename DataType>
+tnsr::I<DataType, 3> LaneEmdenStar::gravitational_field(
+    const tnsr::I<DataType, 3>& x) const noexcept {
+  // Compute alpha for polytrope n==1, units G==1
+  const double alpha = sqrt(0.5 * polytropic_constant_ / M_PI);
+  const double outer_radius = alpha * M_PI;
+  const double mass_scale = 4.0 * M_PI * cube(alpha) * central_mass_density_;
+  // Add tiny offset to avoid divisons by zero
+  const DataType radius = get(magnitude(x)) + 1.e-30 * outer_radius;
+
+  auto enclosed_mass = make_with_value<DataType>(get_size(radius), mass_scale);
+  for (size_t s = 0; s < get_size(radius); ++s) {
+    if (get_element(radius, s) < outer_radius) {
+      const double xi = get_element(radius, s) / alpha;
+      get_element(enclosed_mass, s) *= sin(xi) - xi * cos(xi);
+    } else {
+      get_element(enclosed_mass, s) *= M_PI;
+    }
+  }
+
+  auto gravitational_field_result = x;
+  for (size_t i = 0; i < 3; ++i) {
+    gravitational_field_result.get(i) *= -enclosed_mass / cube(radius);
+  }
+  return gravitational_field_result;
+}
+
+template <typename DataType>
+Scalar<DataType> LaneEmdenStar::precompute_mass_density(
+    const tnsr::I<DataType, 3>& x) const noexcept {
+  // Compute alpha for polytrope n==1, units G==1
+  const double alpha = sqrt(0.5 * polytropic_constant_ / M_PI);
+  const double outer_radius = alpha * M_PI;
+  // Add tiny offset to avoid divisons by zero
+  const DataType radius = get(magnitude(x)) + 1.e-30 * outer_radius;
+
+  Scalar<DataType> mass_density(get_size(radius));
+  for (size_t s = 0; s < get_size(radius); ++s) {
+    if (get_element(radius, s) < outer_radius) {
+      const double xi = get_element(radius, s) / alpha;
+      get_element(get(mass_density), s) = central_mass_density_ * sin(xi) / xi;
+    } else {
+      get_element(get(mass_density), s) = 0.0;
+    }
+  }
+  return mass_density;
+}
+
+template <typename DataType>
+tuples::TaggedTuple<Tags::MassDensity<DataType>> LaneEmdenStar::variables(
+    tmpl::list<Tags::MassDensity<DataType>> /*meta*/,
+    const Scalar<DataType>& mass_density) const noexcept {
+  return mass_density;
+}
+
+template <typename DataType>
+tuples::TaggedTuple<Tags::Velocity<DataType, 3, Frame::Inertial>>
+LaneEmdenStar::variables(
+    tmpl::list<Tags::Velocity<DataType, 3, Frame::Inertial>> /*meta*/,
+    const Scalar<DataType>& mass_density) const noexcept {
+  return make_with_value<tnsr::I<DataType, 3>>(get(mass_density), 0.0);
+}
+
+template <typename DataType>
+tuples::TaggedTuple<Tags::Pressure<DataType>> LaneEmdenStar::variables(
+    tmpl::list<Tags::Pressure<DataType>> /*meta*/,
+    const Scalar<DataType>& mass_density) const noexcept {
+  return Scalar<DataType>{
+      equation_of_state_.pressure_from_density(mass_density)};
```

Comment:
I think the EOS functions return `Scalar`s, so can you remove the `Scalar<DataType>` constructor? (same in function below)